kinds/apis/v1beta1/config_types.go

```diff
@@ -91,32 +91,55 @@ type Chart struct {
 	Values   string `json:"values,omitempty"`
 	TargetNS string `json:"namespace,omitempty"`
 	// Timeout specifies the timeout for how long to wait for the chart installation to finish.
-	// +kubebuilder:validation:Optional
-	Timeout time.Duration `json:"timeout,omitempty"`
+	// A duration string is a sequence of decimal numbers, each with optional fraction and a unit suffix, such as "300ms" or "2h45m". Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+	// +kubebuilder:validation:XIntOrString
+	Timeout BackwardCompatibleDuration `json:"timeout,omitempty"`
 	// ForceUpgrade when set to false, disables the use of the "--force" flag when upgrading the the chart (default: true).
 	// +optional
 	ForceUpgrade *bool `json:"forceUpgrade,omitempty"`
 	// +kubebuilder:validation:Optional
 	Order int `json:"order,omitempty"`
 }
 
-type Repository struct {
-	Name     string `json:"name,omitempty"`
-	URL      string `json:"url,omitempty"`
-	CAFile   string `json:"caFile,omitempty"`
-	CertFile string `json:"certFile,omitempty"`
-	Insecure bool   `json:"insecure,omitempty"`
-	KeyFile  string `json:"keyfile,omitempty"`
-	Username string `json:"username,omitempty"`
-	Password string `json:"password,omitempty"`
+// BackwardCompatibleDuration is a metav1.Duration with a different JSON
+// unmarshaler. The unmashaler accepts its value as either a string (e.g.
+// 10m15s) or as an integer 64. If the value is of type integer then, for
+// backward compatibility, it is interpreted as nano seconds.
+type BackwardCompatibleDuration metav1.Duration
+
+// MarshalJSON marshals the BackwardCompatibleDuration to integer nano seconds,
+// the reverse of the k0s type.
+func (b BackwardCompatibleDuration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(b.Duration)
```

Comment:
subtle difference between the one in k0s

https://github.com/k0sproject/k0s/blob/9a7b4d1fb7b60cdc1b9e13ece2a3ab2e0f4a23c3/pkg/apis/k0s/v1beta1/extensions.go#L134